processor/src/test/java/com/linecorp/decaton/processor/runtime/ProcessorSubscriptionTest.java

```diff
@@ -170,4 +179,49 @@ public void testOffsetRegression() throws Exception {
         OffsetAndMetadata offset = offsets.get(tp);
         assertEquals(100L, offset.offset());
     }
+
+    @Test(timeout = 10000L)
+    public void testTerminateAsync() throws Exception {
+        TopicPartition tp = new TopicPartition("topic", 0);
+        DecatonMockConsumer consumer = new DecatonMockConsumer();
+        for (int i = 0; i < 10; i++) {
```

Comment:
Having hard time to understand why this needs to be repeated for 10 times... can you explain how it works please?

Reply:
The idea was to put enough tasks in the queue that it would call one every time. Let me just implement the right behaviour directly instead.